### PR TITLE
USHIFT-1746: Ansible support prerelease repos

### PR DESCRIPTION
**Pull request by sjug**
**Which issue(s) this PR addresses**:
Closes #[USHIFT-1746](https://issues.redhat.com//browse/USHIFT-1746)

### CODE REVIEW

**Comment by agullon** [comment]
Hi @sjug can you provide more details about how to run it? 
What I understand from reading the code is that I have to override the MicroShift version here: `ansible/roles/manage-repos/defaults/main.yml:4` and check `microshift_build` is set to false in `ansible/vars/all.yml`
Is it correct?

File: ansible/roles/microshift-start/tasks/main.yml [hunk 1]
```diff
@@ -18,7 +23,9 @@
   - name: cleanup microshift data
     become: yes
     ansible.builtin.shell: echo 1 | {{ microshift_cleanup_bin }} --all
-  when: cleanup_microshift | default('true') | bool
+  when:
+    - cleanup_microshift | default('true') | bool
+    - microshift_active.stdout | int != 0
 
 - name: disable microshift service
```
Comment (agullon) [inline ansible/roles/microshift-start/tasks/main.yml:30]: Error reported on this step: `fatal: [microshift-dev]: FAILED! => {"changed": false, "msg": "Could not find the requested service microshift: host"}`
You can check the whole output in Jenkins: https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/microshift/job/microshift-performance-tests/71/console

Comment (agullon) [reply ansible/roles/microshift-start/tasks/main.yml:30]: Rerun it again, skipping the task `disable microshift service`, but the execution is stuck in the task:
`TASK [microshift-start : run the microshift boot script] ***********************`

I ssh into the VM and run:
```
[microshift@microshift-dev ~]$ sudo systemctl status microshift.service
Unit microshift.service could not be found.
```
Jenkins execution log: https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/microshift/job/microshift-performance-tests/72/consoleFull

**Comment by sjug** [comment]
> Hi @sjug can you provide more details about how to run it? What I understand from reading the code is that I have to override the MicroShift version here: `ansible/roles/manage-repos/defaults/main.yml:4` and check `microshift_build` is set to false in `ansible/vars/all.yml` Is it correct?

You don't need to set it in any specific file, you can alternatively override it on the command line.
You need to set `microshift_version` and `microshift_build`.

File: ansible/roles/microshift-start/tasks/main.yml (hunk 1 above)
Comment (sjug) [reply ansible/roles/microshift-start/tasks/main.yml:30]: This means that it didn't install MicroShift. In the logs it shows you've skipped the whole role involved in this PR.

Comment (sjug) [reply ansible/roles/microshift-start/tasks/main.yml:30]: And also the `manage_repos` role, as that's what installs the pre-release repo (or any of them).
For now just set it, and then we can change it in a different PR if that's what we determine.

Comment (agullon) [reply ansible/roles/microshift-start/tasks/main.yml:30]: It seems the mirror repo for pre-releases is failing. Can you take a look, please? 

```
TASK [manage-repos : install EPEL repo] ****************************************
10-11 16:37:46.696  fatal: [microshift-dev]: FAILED! => {"changed": false, "msg": "Failed to download metadata for repo 'microshift-4.14-for-rhel-9-mirrorbeta-x86_64-rpms': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried", "rc": 1, "results": []}
```

https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/microshift/job/microshift-performance-tests/75/console

Comment (sjug) [reply ansible/roles/microshift-start/tasks/main.yml:30]: The logs show you set `microshift_version=4.14.0-rc.3` which is not a version that exists in the repo. There is `4.14.0-rc.2` or  `4.14.0-rc.4`

**Review by agullon — APPROVED** [review]
Tested successfully: https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/microshift/job/microshift-performance-tests/79/console

**Comment by eggfoobar** [comment]
Thanks guys, @sjug I don't have a lot of context on this, would this change require any updates to the `README.md` in the ansible folder?

**Pull request merged**